modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/ReplaceSourceWithDownsampleIndexTask.java

```diff
@@ -0,0 +1,205 @@
+        if (downsampleIndexMeta == null) {
+            // the downsample index doesn't exist anymore so nothing to replace here
+            LOGGER.trace(
+                "Received request replace index [{}] with [{}] in data stream [{}] but the replacement index [{}] doesn't exist."
+                    + "Nothing to do here.",
+                sourceBackingIndex,
+                downsampleIndex,
+                dataStreamName,
+                downsampleIndex
+            );
+            return state;
+        }
+        IndexMetadata sourceIndexMeta = state.metadata().index(sourceBackingIndex);
+        DataStream dataStream = state.metadata().dataStreams().get(dataStreamName);
+        if (sourceIndexAbstraction == null) {
+            // index was deleted in the meantime, so let's check if we can make sure the downsample index ends up in the
+            // data stream (if not already there)
+            if (dataStream != null
+                && dataStream.getIndices().stream().filter(index -> index.getName().equals(downsampleIndex)).findAny().isEmpty()) {
+                // add downsample index to data stream
+                LOGGER.trace(
+                    "unable find source index [{}] but adding index [{}] to data stream [{}]",
+                    sourceBackingIndex,
+                    downsampleIndex,
+                    dataStreamName
+                );
+                Metadata.Builder newMetaData = Metadata.builder(state.metadata())
+                    .put(dataStream.addBackingIndex(state.metadata(), downsampleIndexMeta.getIndex()));
+                return ClusterState.builder(state).metadata(newMetaData).build();
+            }
+        } else {
+            // the source index exists
+            DataStream sourceParentDataStream = sourceIndexAbstraction.getParentDataStream();
+            if (sourceParentDataStream != null) {
+                assert sourceParentDataStream.getName().equals(dataStreamName)
+                    : "the backing index must be part of the provided data "
+                        + "stream ["
+                        + dataStreamName
+                        + "] but it is instead part of data stream ["
+                        + sourceParentDataStream.getName()
+                        + "]";
+                if (sourceParentDataStream.getWriteIndex().getName().equals(sourceBackingIndex)) {
+                    String errorMessage = String.format(
+                        Locale.ROOT,
+                        "index [%s] is the write index for data stream [%s] and cannot be replaced",
+                        sourceBackingIndex,
+                        sourceParentDataStream.getName()
+                    );
+                    throw new IllegalStateException(errorMessage);
+                }
+                if (sourceIndexMeta != null) {
+                    // both indices exist, let's copy the origination date from the source index to the downsample index
```

Comment:
This is important in order for the downsampled index to have a generation date (which would be used as a reference when calculating if the index is eligible for further downsampling rounds)